include/utils.hpp

```diff
@@ -395,6 +396,22 @@ static_assert(std::is_same_v<std::vector<int>, type_transform<std::vector, int>>
 static_assert(std::is_same_v<std::tuple<std::vector<int>, std::vector<float>>, type_transform<std::vector, std::tuple<int, float>>>);
 static_assert(std::is_same_v<void, type_transform<std::vector, void>>);
 
+#ifdef __cpp_lib_hardware_interference_size
+static constexpr const std::size_t kCacheLine = std::hardware_destructive_interference_size;
+#else
+static constexpr const std::size_t kCacheLine = 64;
+#endif
+
+constexpr bool
+isPower2(std::unsigned_integral auto value) {
```

Comment:
It could be -- in theory -- expressed as `std::has_single_bit(value)`. We should check whether this is still needed. Most of the buffers can also work with non-powers-of-twos...

Reply:
Indeed, I have changed it too `std::has_single_bit()` now